comp_fund/MooreTpgm4/MooreTpgm4/MooreTpgm4.1
.\"Modified from man(1) of FreeBSD, the NetBSD mdoc.template, and mdoc.samples.
.\"See Also:
.\"man mdoc.samples for a complete listing of options
.\"man mdoc for the short list of editing options
.\"/usr/share/misc/mdoc.template
.Dd 3/26/14               \" DATE 
.Dt MooreTpgm4 1      \" Program name and manual section number 
.Os Darwin
.Sh NAME                 \" Section Header - required - don't modify 
.Nm MooreTpgm4,
.\" The following lines are read in generating the apropos(man -k) database. Use only key
.\" words here as the database is built based on the words here and in the .ND line. 
.Nm Other_name_for_same_program(),
.Nm Yet another name for the same program.
.\" Use .Nm macro to designate other names for the documented program.
.Nd This line parsed for whatis database.
.Sh SYNOPSIS             \" Section Header - required - don't modify
.Nm
.Op Fl abcd              \" [-abcd]
.Op Fl a Ar path         \" [-a path] 
.Op Ar file              \" [file]
.Op Ar                   \" [file ...]
.Ar arg0                 \" Underlined argument - use .Ar anywhere to underline
arg2 ...                 \" Arguments
.Sh DESCRIPTION          \" Section Header - required - don't modify
Use the .Nm macro to refer to your program throughout the man page like such:
.Nm
Underlining is accomplished with the .Ar macro like this:
.Ar underlined text .
.Pp                      \" Inserts a space
A list of items with descriptions:
.Bl -tag -width -indent  \" Begins a tagged list 
.It item a               \" Each item preceded by .It macro
Description of item a
.It item b
Description of item b
.El                      \" Ends the list
.Pp
A list of flags and their descriptions:
.Bl -tag -width -indent  \" Differs from above in tag removed 
.It Fl a                 \"-a flag as a list item
Description of -a flag
.It Fl b
Description of -b flag
.El                      \" Ends the list
.Pp
.\" .Sh ENVIRONMENT      \" May not be needed
.\" .Bl -tag -width "ENV_VAR_1" -indent \" ENV_VAR_1 is width of the string ENV_VAR_1
.\" .It Ev ENV_VAR_1
.\" Description of ENV_VAR_1
.\" .It Ev ENV_VAR_2
.\" Description of ENV_VAR_2
.\" .El                      
.Sh FILES                \" File used or created by the topic of the man page
.Bl -tag -width "/Users/joeuser/Library/really_long_file_name" -compact
.It Pa /usr/share/file_name
FILE_1 description
.It Pa /Users/joeuser/Library/really_long_file_name
FILE_2 description
.El                      \" Ends the list
.\" .Sh DIAGNOSTICS       \" May not be needed
.\" .Bl -diag
.\" .It Diagnostic Tag
.\" Diagnostic informtion here.
.\" .It Diagnostic Tag
.\" Diagnostic informtion here.
.\" .El
.Sh SEE ALSO 
.\" List links in ascending order by section, alphabetically within a section.
.\" Please do not reference files that do not exist without filing a bug report
.Xr a 1 , 
.Xr b 1 ,
.Xr c 1 ,
.Xr a 2 ,
.Xr b 2 ,
.Xr a 3 ,
.Xr b 3 
.\" .Sh BUGS              \" Document known, unremedied bugs 
.\" .Sh HISTORY           \" Document history if command behaves in a unique manner

comp_fund/MooreTpgm4/MooreTpgm4/MooreTpgm4.c
//
//  MooreTpgm4.c
//
//  Created by Tiago Moore on 3/26/14.
//  Copyright (c) 2014 T Moore. All rights reserved.
/*********************************************************************
 Author    : Tiago Moore
 Course    : CDA3101 MWF 1300-1350
 Professor : Michael Robinson
 Program # : 4  purpose is to to Cover chapters 13 and 14 using C language
 
 Due Date : 04/16/2014
 
 Certification:
 I hereby certify that this work is my own and none of it is the work of any other person.
 ..........{ your signature }..........
 *********************************************************************/
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
// funtion prototypes 

// return type: void
// Parameters: 1; the file to be read
// This function reads the DNA file into a a memory location(pointer) containing an character array
//void readFile( char *a[]);

// return type : void
// Parameters 2; the genome sequence( char array), and the number of nuecleotides
// Purpose:This function counts teh amount of nucleotides in the file, the first, last middle,
// as well as the amount for each nucleotide lowercase, uppercase, and other.
void countTides(char*nucleotide, int numberofTides);

// return type : int
// Parameters: 1 the number to add to
// Purpose: This function will add all numbers from 0 to 100 using a formula
int SumOfDigits(int n);

// return type: void
// Parameters: 1 the file being manipulated
// Purpose: Additional practice for manipulating text files
void createTextFile(FILE *f);

// return type: void
// Parameters: 1 the binary file being manipulated
// Purpose: This function manipulates a binary file
void createBinaryFile(FILE *f);

int main()
{
    // declare and initialize variables
    FILE *dnaFile ;    // my file pointer
    int dataSize =0, i = 0 ;
    char *dnaArray;
    // read in DNA file
    dnaFile = fopen("1200.dna", "r");
    int numToAddTo = 100;

    if(dnaFile == NULL)
    {
        printf("Cannot open the file ");
        exit(0);
    }
     char nextChar  = getc(dnaFile); // character for traversing the file
    // get size of data file
    while(nextChar != EOF)
    {
        dataSize++;
        nextChar = getc(dnaFile);
    }
    // close file
    fclose(dnaFile);
    
   //create an  array in heap the size of data
    dnaArray = malloc(sizeof(int)*dataSize);
    
    //reopen file
    dnaFile = fopen("1200.dna", "r");
    // check for sucess
    if(dnaFile == NULL)
    {
        printf("Cannot open the file ");
        exit(0);
    }
    nextChar  = getc(dnaFile); // character for traversing the file
  // read every char into the array just created with sufficient memory
    while(nextChar != EOF)
    {
        dnaArray[i] = nextChar;
        i++;
        nextChar = getc(dnaFile);
    }
    // close file
    fclose(dnaFile);
    // --------------------------------------- done reading file
    // call function to count everything
    countTides(dnaArray, dataSize);
    
    printf("The sum of the digits from 1 to 100 is %d\n",SumOfDigits(numToAddTo));
    
    FILE *textFile;
    FILE *binaryFile;
    
    createTextFile(textFile);
    createBinaryFile(binaryFile);
    
    return 0;
}

void countTides(char * nuecleotide, int numberofTides)
    {
    int upperA = 0, lowerA = 0, upperC = 0, lowerC = 0, upperT = 0 , lowerT = 0 , upperG = 0 , lowerG = 0, other = 0 ,count = 0;
    char current;
    // print total, first ,last middle
    printf("There are %d nuecluotides in the file.\n",numberofTides);
    printf("The first nueclueotide in the file is : %c\n", nuecleotide[0]);
    printf("The last nueclueotide in the file is : %c\n", nuecleotide[numberofTides-2]);
    printf("The nueclueotide  exacly in the middle is : %c\n", nuecleotide[(int)ceil(numberofTides/2)]) ;// use ceiling function
    // print individual nucleotides
    while( count != numberofTides)
    {
        // get the current nucleotide
        current = nuecleotide[count];
        
        if( current == 'A')
        {
            upperA ++;
        }
        else if( current == 'C')
        {
            upperC++;
        }
        else if(current == 'T')
        {
            upperT++;
        }
        else if( current == 'G')
        {
            upperG++;
        }
        else if( current == 'a')
        {
            lowerA++;
        }
        else if(current == 't')
        {
            lowerT++;
        }
        else if( current == 'g')
        {
            lowerG++;
        }
        else if (current == 'c')
        {
            lowerC++;
        }
        else
        {
            other ++;
        }
    
        count++;
    }
    
    printf("There are:\n%d A nucleotides\t%d a nucleotides\n%d C nucleotides\t%d c nucleotides\n%d G nucleotides\t%d g nucleotides\n%d T nucleotides\t%d t nucleotides\n",upperA,lowerA,upperC,lowerC,upperG,lowerG,upperT,lowerT);
        
        printf("There are %d 'other' characters in the file.\n",other );
}

int SumOfDigits(int n)
{
    //declare and initiliaze variables
    int total = 0;
    // apply the formula
    //(1 plus N quantity) times (N divided by 2)
    total = (n+1)*(n/2);
    return total;
}

void createTextFile(FILE *f)
{
    int count = 0;
    f = fopen("text.txt", "w");
    
    // write to the file one number per record
    for (count = 1;count<=5;count++)
    {
        fprintf(f,"%d\n",count);
    }
    // close file
    fclose(f);
    
    // open file for reading
    f = fopen("text.txt", "r");
    
    if(f == NULL)
    {
        printf("Cannot open text.txt");
        exit(0);
    }
    else
    {
        printf("File text.txt created\n");
    }
    // funciton gets a file pointer and returns an int reading a single char from file
    int  a = fgetc(f);
    while(a != EOF)// while not end of file
    {
        printf("%c",a);
        a = fgetc(f);
        if(a == '\n')  // if a is a new line character,
        a =' ';//replace it with a space
    }
    fclose(f);
    
}
    
void createBinaryFile(FILE *f)
{
    int i = 0;

    printf("\nBinary File binText.bin created");
    
    f = fopen("binText.bin", "wb");// open file for output
    if( f == NULL)
    {
        printf("Cannot open binText.bin");
        exit(0);
    }

    char *letters = "a\nb\nc\nd\ne";// declare char with letters
    
    fwrite(letters,sizeof(char),12,f); // write one letter of the char array per line
    
    fclose(f); // close file

    f = fopen("binText.bin", "rb+");// open file for reading
    if( f == NULL)
    {
        printf("Cannot open binText.bin");
        exit(0);
    }
    
    int j = 0;
    for(j = 0; j < 9 ; j++)
    {
        fread(&i,sizeof(char),1,f); // read each charater
        if(i == 10)// if its a new line
        {
        printf(" "); // replace it with a space
        }
        else
        {
        printf("%c",i);// print the character
        }
}
 
    printf("\n");
    
    fclose(f);
}



    
    
    
    

